torch/quantization/fake_quantize.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch
+from torch.nn import Module
+from .._jit_internal import weak_module
+from .observer import Observer, observer
+
+@weak_module
+class FakeQuantize(Module):
+    ''' Simulate the quantize and dequantize operations in training time.
+    Args:
+        `qconfig`: object that encodes configuration info for quantization
+        `observer_module`: Observer module that records stats of weights and
+        activations
+        `calcqparam`: A function that calculates quantization parameters
+        given the stats
+    '''
+
+    def __init__(self, qconfig, quant_min=-128, quant_max=127, quant_delay=0):
+        super(FakeQuantize, self).__init__()
+        self.observer = qconfig.activation()
+        self.qconfig = qconfig
+        self.quant_min = quant_min
+        self.quant_max = quant_max
+        self.quant_delay = quant_delay
+        self.iter = 1
+        self.scale = None
+        self.zero_point = None
+
+    def forward(self, X):
```

Comment:
should all this logic be disabled in eval mode or not? (i.e. `if self.training`). Not sure what is a reasonable default.

Reply:
we probably don't have FakeQuantize in evaluation, but yeah in eval mode this should be identity.

Reply:
torch.quantization.convert() would remove this logic. We may want to have fake quant in eval mode: To model quantization error without actually writing kernels. If user wants to remove fake quant module in eval mode, then a simple module swap: Replacing this module with identity should do the job